### ISSUE TITLE
White Screen on Signed APK only

### ISSUE THREAD

**Issue by ODelibalta** [wix/react-native-navigation#2447]
### Issue Description

Its been two days since I've been on this issue I am literally pulling my hair out. It used to work, now it does not. I am not sure why. 

### entire log from logcat for my app
```
12-30 14:00:50.530 14332-14332/com.vivaa V/fb-UnpackingSoSource: locked dso store /data/user/0/com.vivaa/lib-main
12-30 14:00:50.532 14332-14332/com.vivaa V/fb-UnpackingSoSource: deps mismatch on deps store: regenerating
12-30 14:00:50.532 14332-14332/com.vivaa V/fb-UnpackingSoSource: so store dirty: regenerating
12-30 14:00:50.591 14332-14332/com.vivaa D/ApkSoSource: not allowing consideration of lib/armeabi-v7a/libcardioDecider.so: deferring to libdir
12-30 14:00:50.591 14332-14332/com.vivaa D/ApkSoSource: not allowing consideration of lib/armeabi-v7a/libcardioRecognizer.so: deferring to libdir
12-30 14:00:50.591 14332-14332/com.vivaa D/ApkSoSource: not allowing consideration of lib/armeabi-v7a/libcardioRecognizer_tegra2.so: deferring to libdir
12-30 14:00:50.591 14332-14332/com.vivaa D/ApkSoSource: not allowing consideration of lib/armeabi-v7a/libfb.so: deferring to libdir
12-30 14:00:50.591 14332-14332/com.vivaa D/ApkSoSource: not allowing consideration of lib/armeabi-v7a/libfolly_json.so: deferring to libdir
12-30 14:00:50.591 14332-14332/com.vivaa D/ApkSoSource: not allowing consideration of lib/armeabi-v7a/libglog.so: deferring to libdir
12-30 14:00:50.591 14332-14332/com.vivaa D/ApkSoSource: not allowing consideration of lib/armeabi-v7a/libglog_init.so: deferring to libdir
12-30 14:00:50.591 14332-14332/com.vivaa D/ApkSoSource: not allowing consideration of lib/armeabi-v7a/libgnustl_shared.so: deferring to libdir
12-30 14:00:50.592 14332-14332/com.vivaa D/ApkSoSource: not allowing consideration of lib/armeabi-v7a/libicu_common.so: deferring to libdir
12-30 14:00:50.592 14332-14332/com.vivaa D/ApkSoSource: not allowing consideration of lib/armeabi-v7a/libimagepipeline.so: deferring to libdir
12-30 14:00:50.592 14332-14332/com.vivaa D/ApkSoSource: not allowing consideration of lib/armeabi-v7a/libjsc.so: deferring to libdir
12-30 14:00:50.592 14332-14332/com.vivaa D/ApkSoSource: not allowing consideration of lib/armeabi-v7a/libopencv_core.so: deferring to libdir
12-30 14:00:50.592 14332-14332/com.vivaa D/ApkSoSource: not allowing consideration of lib/armeabi-v7a/libopencv_imgproc.so: deferring to libdir
12-30 14:00:50.592 14332-14332/com.vivaa D/ApkSoSource: not allowing consideration of lib/armeabi-v7a/libprivatedata.so: deferring to libdir
12-30 14:00:50.592 14332-14332/com.vivaa D/ApkSoSource: not allowing consideration of lib/armeabi-v7a/libreactnativejni.so: deferring to libdir
12-30 14:00:50.592 14332-14332/com.vivaa D/ApkSoSource: not allowing consideration of lib/armeabi-v7a/libyoga.so: deferring to libdir
12-30 14:00:50.592 14332-14332/com.vivaa V/fb-UnpackingSoSource: regenerating DSO store com.facebook.soloader.ApkSoSource
12-30 14:00:50.593 14332-14332/com.vivaa V/fb-UnpackingSoSource: starting syncer worker
12-30 14:00:50.602 14332-14332/com.vivaa V/fb-UnpackingSoSource: releasing dso store lock for /data/user/0/com.vivaa/lib-main (from syncer thread)
12-30 14:00:50.602 14332-14332/com.vivaa V/fb-UnpackingSoSource: not releasing dso store lock for /data/user/0/com.vivaa/lib-main (syncer thread started)
12-30 14:00:50.616 14332-14332/com.vivaa W/art: Before Android 4.1, method android.graphics.PorterDuffColorFilter android.support.graphics.drawable.VectorDrawableCompat.updateTintFilter(android.graphics.PorterDuffColorFilter, android.content.res.ColorStateList, android.graphics.PorterDuff$Mode) would have incorrectly overridden the package-private method in android.graphics.drawable.Drawable
12-30 14:00:50.639 14332-14332/com.vivaa D/ReactNative: ReactInstanceManager.ctor()
12-30 14:00:50.642 14332-14332/com.vivaa D/ReactNative: ReactInstanceManager.createReactContextInBackground()
12-30 14:00:50.643 14332-14332/com.vivaa D/ReactNative: ReactInstanceManager.recreateReactContextInBackgroundInner()
12-30 14:00:50.643 14332-14332/com.vivaa D/ReactNative: ReactInstanceManager.recreateReactContextInBackgroundFromBundleLoader()
12-30 14:00:50.643 14332-14332/com.vivaa D/ReactNative: ReactInstanceManager.recreateReactContextInBackground()
12-30 14:00:50.643 14332-14332/com.vivaa D/ReactNative: ReactInstanceManager.runCreateReactContextOnNewThread()
12-30 14:00:50.671 14332-14349/com.vivaa I/art: Thread[12,tid=14349,Native,Thread*=0xb3db9800,peer=0x12dfd8b0,"Thread-2"] recursive attempt to load library "/data/app/com.vivaa-1/lib/arm/libfb.so"
12-30 14:00:50.673 14332-14349/com.vivaa D/ReactNative: ReactInstanceManager.createReactContext()
12-30 14:00:50.688 14332-14349/com.vivaa D/React:  View manager getCommandsMap:
12-30 14:00:50.689 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.rssignaturecapture.RSSignatureCaptureViewManager
12-30 14:00:50.704 14332-14350/com.vivaa I/Adreno-EGL: <qeglDrvAPI_eglInitialize:379>: EGL 1.4 QUALCOMM build: Nondeterministic_AU_msm8974_LA.BF.1.1.3_RB1__release_AU (Ia6c73e7530)
                                                       OpenGL ES Shader Compiler Version: E031.29.00.00
                                                       Build Date: 11/17/15 Tue
                                                       Local Branch: mybranch16463693
                                                       Remote Branch: quic/LA.BF.1.1.3_rb1.4
                                                       Local Patches: NONE
                                                       Reconstruct Branch: NOTHING
12-30 14:00:50.709 14332-14350/com.vivaa I/OpenGLRenderer: Initialized EGL, version 1.4
12-30 14:00:50.710 14332-14350/com.vivaa D/OpenGLRenderer: Swap behavior 1
12-30 14:00:50.716 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.uimanager.LayoutShadowNode
12-30 14:00:50.725 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.lwansbrough.RCTCamera.RCTCameraViewManager
12-30 14:00:50.729 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.art.ARTGroupViewManager
12-30 14:00:50.730 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.art.ARTGroupShadowNode
12-30 14:00:50.731 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.art.ARTShapeViewManager
12-30 14:00:50.732 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.art.ARTShapeShadowNode
12-30 14:00:50.735 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.art.ARTTextViewManager
12-30 14:00:50.735 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.art.ARTTextShadowNode
12-30 14:00:50.736 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.checkbox.ReactCheckBoxManager
12-30 14:00:50.738 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.picker.ReactDialogPickerManager
12-30 14:00:50.741 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.drawer.ReactDrawerLayoutManager
12-30 14:00:50.746 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.picker.ReactDropdownPickerManager
12-30 14:00:50.747 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.scroll.ReactHorizontalScrollViewManager
12-30 14:00:50.753 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.scroll.ReactHorizontalScrollContainerViewManager
12-30 14:00:50.754 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.progressbar.ReactProgressBarViewManager
12-30 14:00:50.756 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.progressbar.ProgressBarShadowNode
12-30 14:00:50.758 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.scroll.ReactScrollViewManager
12-30 14:00:50.762 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.slider.ReactSliderManager
12-30 14:00:50.765 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.slider.ReactSliderManager$ReactSliderShadowNode
12-30 14:00:50.766 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.switchview.ReactSwitchManager
12-30 14:00:50.769 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.switchview.ReactSwitchManager$ReactSwitchShadowNode
12-30 14:00:50.769 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.toolbar.ReactToolbarManager
12-30 14:00:50.773 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.webview.ReactWebViewManager
12-30 14:00:50.777 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.swiperefresh.SwipeRefreshLayoutManager
12-30 14:00:50.781 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.art.ARTSurfaceViewManager
12-30 14:00:50.781 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.art.ARTSurfaceViewShadowNode
12-30 14:00:50.782 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.text.frescosupport.FrescoBasedReactTextInlineImageViewManager
12-30 14:00:50.783 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.text.frescosupport.FrescoBasedReactTextInlineImageShadowNode
12-30 14:00:50.785 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.image.ReactImageManager
12-30 14:00:50.789 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.modal.ReactModalHostManager
12-30 14:00:50.791 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.modal.ModalHostShadowNode
12-30 14:00:50.792 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.text.ReactRawTextManager
12-30 14:00:50.793 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.text.ReactRawTextShadowNode
12-30 14:00:50.794 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.textinput.ReactTextInputManager
12-30 14:00:50.801 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.textinput.ReactTextInputShadowNode
12-30 14:00:50.805 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.text.ReactTextViewManager
12-30 14:00:50.809 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.text.ReactTextShadowNode
12-30 14:00:50.809 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.view.ReactViewManager
12-30 14:00:50.815 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.viewpager.ReactViewPagerManager
12-30 14:00:50.818 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.text.ReactVirtualTextViewManager
12-30 14:00:50.819 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.text.ReactVirtualTextShadowNode
12-30 14:00:50.819 14332-14349/com.vivaa W/unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.reactnativenavigation.views.managers.SharedElementTransitionManager
12-30 14:00:50.852 14332-14349/com.vivaa D/NetworkSecurityConfig: No Network Security Config specified, using platform default
12-30 14:00:50.856 14332-14358/com.vivaa I/OMXClient: MuxOMX ctor
12-30 14:00:50.860 14332-14349/com.vivaa D/ReactNative: Initializing React Xplat Bridge.
12-30 14:00:50.862 14332-14349/com.vivaa D/ReactNative: Initializing React Xplat Bridge before initializeBridge
12-30 14:00:50.876 14332-14349/com.vivaa D/ReactNative: Initializing React Xplat Bridge after initializeBridge
12-30 14:00:50.876 14332-14349/com.vivaa D/ReactNative: CatalystInstanceImpl.runJSBundle()
12-30 14:00:50.891 14332-14362/com.vivaa D/ReactNative: ReactInstanceManager.setupReactContext()
12-30 14:00:50.891 14332-14362/com.vivaa D/ReactNative: CatalystInstanceImpl.initialize()
```
On a non signed build, app runs fine because after the CatalystInstance there is a ``` ReactInstanceManager.attachRootViewToInstance()``` line but it only happens on non release builds.

### package.json
```  "dependencies": {
    "axios": "^0.17.1",
    "date-fns": "^1.29.0",
    "lodash": "^4.17.4",
    "native-base": "^2.3.5",
    "prop-types": "^15.6.0",
    "react": "16.0.0",
    "react-native": "0.51.0",
    "react-native-awesome-card-io": "^0.7.0",
    "react-native-camera": "^0.12.0",
    "react-native-easy-grid": "^0.1.15",
    "react-native-elements": "^0.18.5",
    "react-native-modal": "^4.1.1",
    "react-native-modal-datetime-picker": "^4.13.0",
    "react-native-navigation": "^1.1.323",
    "react-native-signature-capture": "^0.4.9",
    "react-native-vector-icons": "^4.4.2",
    "react-redux": "^5.0.6",
    "redux": "^3.7.2",
    "redux-persist": "^4.8.2",
    "redux-thunk": "^2.2.0"
  },
```

### app.android.js
```
import React from 'react';
import { Navigation } from 'react-native-navigation';
import { Provider } from 'react-redux';
import { registerScreens } from './screens';
import store from './store';
import { navBarStyle, drawerStyle } from './utils/AppIosStyles';

registerScreens(store, Provider); // this is where you register all of your app's screens

const homeIcon = require('./images/home.png');
const appointmentIcon = require('./images/calendar-text.png');
const customersIcon = require('./images/clipboard-account.png');
const viewTurn = require('./images/update.png');
const availabilityIcon = require('./images/human-greeting.png');

// start the app
class App extends React.Component {
  constructor(props) {
    super(props);

    store.subscribe(this.onStoreUpdate.bind(this));

    this.onStoreUpdate.bind(this);
  }

  onStoreUpdate() {
    if (this.currentSuccess !== store.getState().auth.canUseApp) {
      this.currentSuccess = store.getState().auth.canUseApp;
      if (!this.currentSuccess) {
        App.startApp('single');
      } else {
        App.startApp('tab');
      }
    }
  }

  static startApp(navType) {
    const drawerButton = {
      id: 'sideMenu',
      disableIconTint: true,
      buttonFontSize: 14,
      buttonFontWeight: '600',
    };

    if (navType === 'single') {
      try {
        Navigation.startSingleScreenApp({
          screen: {
            screen: 'vivaa.LoginScreen',
            title: 'Login',
            navigatorStyle: {
              navBarHidden: true,
            },
          },
        });
      } catch (e) {
        console.log(e);
      }
    } else {
      try {
        Navigation.startTabBasedApp({
          tabs: [
            {
              label: 'Home',
              screen: 'vivaa.HomeTabScreen', // this is a registered name for a screen
              icon: homeIcon,
              title: 'Vivaa Home',
              navigatorButtons: {
                leftButtons: [drawerButton],
              },
              navigatorStyle: navBarStyle,
            },
            {
              label: 'Appointments',
              screen: 'vivaa.AppointmentsTabScreen',
              icon: appointmentIcon,
              title: 'Appointments',
              navigatorButtons: {
                leftButtons: [drawerButton],
              },
              navigatorStyle: navBarStyle,
            },
            {
              label: 'Customers',
              screen: 'vivaa.CustomersTabScreen',
              icon: customersIcon,
              title: 'Customers',
              navigatorButtons: {
                leftButtons: [drawerButton],
              },
              navigatorStyle: navBarStyle,
            },
            {
              label: 'View Turn',
              screen: 'vivaa.ViewTurnTabScreen',
              icon: viewTurn,
              title: 'View Turn',
              navigatorButtons: {
                leftButtons: [drawerButton],
              },
              navigatorStyle: navBarStyle,
            },
            {
              label: 'Check Availability',
              screen: 'vivaa.CheckAvailabilityTabScreen',
              icon: availabilityIcon,
              title: 'Check Availability',
              navigatorButtons: {
                leftButtons: [drawerButton],
              },
              navigatorStyle: navBarStyle,
            },
          ],
          drawer: {
            left: {
              screen: 'vivaa.SideMenuScreen',
              passProps: {},
            },
            style: drawerStyle,
            disableOpenGesture: false,
          },
          appStyle: {
            forceTitlesDisplay: true,
          },
        });
        // */
      } catch (e) {
        console.log(e);
      }
    }
  }
}

export default App;
```

This all used to work just fine I do not know what changed.

**Comment by ODelibalta** [comment]
I have taken off everything from my android js to the point where it just returns the default component when you do a fresh react native install and I still get the same result.

**Comment by ODelibalta** [comment]
Just did a fresh react native project install. Started installing my libs from scratch. After each npm install, I did a release build and it worked until I installed this lib. I am really going crazy. I can not wait to finish my current project so I never use this lib again. It has been a major pain too bad I am way into my current project. No errors just white screen on a fresh install with nothing but libs listed in my package.json above.

**Comment by pentarex** [comment]
you said everything worked before only on signed app is not... how are you installing it in your java file? the newer versions of react-native (above 0.49) doesn't have index.android.js anymore. so you should point to index.js. Also try some simple example like in index.js to return only singleScreenApp and see if something happens. You can also specify the version of the library you are using.

**Comment by ODelibalta** [comment]
@pentarex thank you so much for your reply. 

My version along with every other lib is listed in my package.json above. 
I have done the simplest install that I can with creating a brand new react native app with react native init ... then touching nothing but installing the libs one by one. After each install, I did a release apk and it opened the welcome to react native default screen each time till I installed the nav library. 

### settings.gradle
```
rootProject.name = 'vivaa'
include ':app'
include ':react-native-signature-capture'
project(':react-native-signature-capture').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-signature-capture/android')
include ':react-native-camera'
project(':react-native-camera').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-camera/android')
include ':react-native-awesome-card-io'
project(':react-native-awesome-card-io').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-awesome-card-io/android')
include ':react-native-vector-icons'
project(':react-native-vector-icons').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-vector-icons/android')
include ':react-native-navigation'
project(':react-native-navigation').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-navigation/android/app/')
```

### build.gradle
```
apply plugin: "com.android.application"

import com.android.build.OutputFile

project.ext.react = [
    entryFile: "index.js"
]

apply from: "../../node_modules/react-native/react.gradle"
apply from: "../../node_modules/react-native-vector-icons/fonts.gradle"

/**
 * Set this to true to create two separate APKs instead of one:
 *   - An APK that only works on ARM devices
 *   - An APK that only works on x86 devices
 * The advantage is the size of the APK is reduced by about 4MB.
 * Upload all the APKs to the Play Store and people will download
 * the correct one based on the CPU architecture of their device.
 */
def enableSeparateBuildPerCPUArchitecture = false

/**
 * Run Proguard to shrink the Java bytecode in release builds.
 */
def enableProguardInReleaseBuilds = false

android {
    compileSdkVersion 25
    buildToolsVersion "25.0.3"

    defaultConfig {
        applicationId "com.vivaa"
        minSdkVersion 16
        targetSdkVersion 22
        versionCode 1
        versionName "1.0"
        ndk {
            abiFilters "armeabi-v7a", "x86"
        }
    }
    signingConfigs {
        release {
            if (project.hasProperty('MYAPP_RELEASE_STORE_FILE')) {
                storeFile file(MYAPP_RELEASE_STORE_FILE)
                storePassword MYAPP_RELEASE_STORE_PASSWORD
                keyAlias MYAPP_RELEASE_KEY_ALIAS
                keyPassword MYAPP_RELEASE_KEY_PASSWORD
            }
        }
    }
    splits {
        abi {
            reset()
            enable enableSeparateBuildPerCPUArchitecture
            universalApk false  // If true, also generate a universal APK
            include "armeabi-v7a", "x86"
        }
    }
    buildTypes {
        release {
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
            signingConfig signingConfigs.release
        }
    }
    // applicationVariants are e.g. debug, release
    applicationVariants.all { variant ->
        variant.outputs.each { output ->
            // For each separate APK per architecture, set a unique version code as described here:
            // http://tools.android.com/tech-docs/new-build-system/user-guide/apk-splits
            def versionCodes = ["armeabi-v7a":1, "x86":2]
            def abi = output.getFilter(OutputFile.ABI)
            if (abi != null) {  // null for the universal-debug, universal-release variants
                output.versionCodeOverride =
                        versionCodes.get(abi) * 1048576 + defaultConfig.versionCode
            }
        }
    }
}

dependencies {
    compile fileTree(dir: "libs", include: ["*.jar"])
    compile "com.android.support:appcompat-v7:23.0.1"
    compile "com.facebook.react:react-native:+"  // From node_modules
    compile project(':react-native-signature-capture')
    compile project(':react-native-camera')
    compile project(':react-native-awesome-card-io')
    compile project(':react-native-vector-icons')
    compile project(':react-native-navigation')
}

// Run this once to be able to run the application with BUCK
// puts all compile dependencies into folder libs for BUCK to use
task copyDownloadableDepsToLibs(type: Copy) {
    from configurations.compile
    into 'libs'
}
```

### mainapplication 

```
package com.vivaa;

import android.app.Application;
import android.support.annotation.NonNull;

import com.facebook.react.ReactApplication;
import com.rssignaturecapture.RSSignatureCapturePackage;
import com.lwansbrough.RCTCamera.RCTCameraPackage;
import com.cardio.RNCardIOPackage;
import com.oblador.vectoricons.VectorIconsPackage;
import com.reactnativenavigation.NavigationApplication;
import com.facebook.react.ReactNativeHost;
import com.facebook.react.ReactPackage;
import com.facebook.react.shell.MainReactPackage;
import com.facebook.soloader.SoLoader;

import java.util.Arrays;
import java.util.List;

public class MainApplication extends NavigationApplication {
  @Override
  public boolean isDebug() {
    // Make sure you are using BuildConfig from your own application
    return BuildConfig.DEBUG;
  }

  @NonNull
  @Override
  public List<ReactPackage> createAdditionalReactPackages() {
    return Arrays.<ReactPackage>asList(
      // new MainReactPackage(),
      new RSSignatureCapturePackage(),
      new RCTCameraPackage(),
      new RNCardIOPackage(),
      new VectorIconsPackage()
    );
  }


  @Override
  public void onCreate() {
    super.onCreate();
    SoLoader.init(this, /* native exopackage */ false);
  }

  @Override
  public String getJSMainModuleName() {
      return "index";
  }
}
```


### index.js
```
import { AppRegistry, Platform } from 'react-native';
import IosApp from './src/app.ios';
import AndroidApp from './src/app.android';

if (Platform.OS === 'ios') {
  AppRegistry.registerComponent('vivaa', () => IosApp);
} else {
  AppRegistry.registerComponent('vivaa', () => AndroidApp);
}
```

### app.android.js

```
import React from 'react';
import { Navigation } from 'react-native-navigation';
import { Provider } from 'react-redux';
import { registerScreens } from './screens';
import store from './store';
import { navBarStyle, drawerStyle } from './utils/AppIosStyles';

registerScreens(store, Provider); // this is where you register all of your app's screens

const homeIcon = require('./images/home.png');
const appointmentIcon = require('./images/calendar-text.png');
const customersIcon = require('./images/clipboard-account.png');
const viewTurn = require('./images/update.png');
const availabilityIcon = require('./images/human-greeting.png');

// start the app
class App extends React.Component {
  constructor(props) {
    super(props);

    store.subscribe(this.onStoreUpdate.bind(this));

    this.onStoreUpdate.bind(this);
    console.log('this is constructor at the end');
  }

  onStoreUpdate() {
    console.log('this is on store update');
    if (this.currentSuccess !== store.getState().auth.canUseApp) {
      this.currentSuccess = store.getState().auth.canUseApp;
      if (!this.currentSuccess) {
        App.startApp('single');
      } else {
        App.startApp('tab');
      }
    }
  }

  static startApp(navType) {
    console.log('this is start app');
    const drawerButton = {
      id: 'sideMenu',
      disableIconTint: true,
      buttonFontSize: 14,
      buttonFontWeight: '600',
    };

    if (navType === 'single') {
      try {
        Navigation.startSingleScreenApp({
          screen: {
            screen: 'vivaa.LoginScreen',
            title: 'Login',
            navigatorStyle: {
              navBarHidden: true,
            },
          },
        });
      } catch (e) {
        console.log(e);
      }
    } else {
      try {
        Navigation.startTabBasedApp({
          tabs: [
            {
              label: 'Home',
              screen: 'vivaa.HomeTabScreen', // this is a registered name for a screen
              icon: homeIcon,
              title: 'Vivaa Home',
              navigatorButtons: {
                leftButtons: [drawerButton],
              },
              navigatorStyle: navBarStyle,
            },
            {
              label: 'Appointments',
              screen: 'vivaa.AppointmentsTabScreen',
              icon: appointmentIcon,
              title: 'Appointments',
              navigatorButtons: {
                leftButtons: [drawerButton],
              },
              navigatorStyle: navBarStyle,
            },
            {
              label: 'Customers',
              screen: 'vivaa.CustomersTabScreen',
              icon: customersIcon,
              title: 'Customers',
              navigatorButtons: {
                leftButtons: [drawerButton],
              },
              navigatorStyle: navBarStyle,
            },
            {
              label: 'View Turn',
              screen: 'vivaa.ViewTurnTabScreen',
              icon: viewTurn,
              title: 'View Turn',
              navigatorButtons: {
                leftButtons: [drawerButton],
              },
              navigatorStyle: navBarStyle,
            },
            {
              label: 'Check Availability',
              screen: 'vivaa.CheckAvailabilityTabScreen',
              icon: availabilityIcon,
              title: 'Check Availability',
              navigatorButtons: {
                leftButtons: [drawerButton],
              },
              navigatorStyle: navBarStyle,
            },
          ],
          drawer: {
            left: {
              screen: 'vivaa.SideMenuScreen',
              passProps: {},
            },
            style: drawerStyle,
            disableOpenGesture: false,
          },
          appStyle: {
            forceTitlesDisplay: true,
          },
        });
        // */
      } catch (e) {
        console.log(e);
      }
    }
  }
}

export default App;
```

**Comment by guyca** [comment]
Your index.js file is a bit messed up.
Why do you require both variants of app.js? The packager resolves the correct file when bundling your app for each platform.
You don't need to call AppRegistry.registerComponent.

your index file should be something along these lines:

```js
const App = require('./App');
App();
```

I suggest adding logs to Js side as well...

**Comment by pentarex** [comment]
yeah and if possible try to simplify your app.android.js as much as possible at least for the test (remove the redux statements) and just show a singleScreenApp from the beginning. If still not working, later in the evening I will try to make a repo with a simple redux, persist, rnn example for you. But before try to impl the changes mentioned from @guyca and let us know. Off topic - trust me - stay with this library. I am coming from react-navigation and it is pain, big, big, big pain 🗡  :D

**Comment by pentarex** [comment]
@ODelibalta can you try this if it executes in prod release.

https://github.com/pentarex/React-Native-Navigation_Redux_Persist

**Comment by ODelibalta** [comment]
Guys ... I really appreciate you taking your time to take a look at my problem. 

@guyca you were spot on. My index.js was unnecessarily complicated. I swear I found that somewhere on stackoverflow when I upgrade from 46 to 50 and I only had index.js instead of the two index files. I also saw that you fixed the top root issue on android lightbox. That makes me so happy.

@pentarex Thank you so much. I appreciate your help immensely. I did clear out my app android js to the point where it just fired a single screen app and it worked. That steered me into the correct path. Yes I do love this lib and I can not wait for v2. It is like having a child that becomes super annoying at times but you know you have deep love and appreciation at the end :) 

I am closing this as it was caused by my own dumbassery and I sincerely thank everyone who stopped by and checked it out. I can finally move on. Thank you all
